client/components/notice/index.tsx

```diff
@@ -0,0 +1,149 @@
+/**
+ * External dependencies
+ */
+
+import React, { useEffect, FunctionComponent } from 'react';
+import classnames from 'classnames';
+import { localize } from 'i18n-calypso';
+import Gridicon from 'calypso/components/gridicon';
+
+/**
+ * Internal dependencies
+ */
+import { TimerHandle } from 'calypso/types';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+/**
+ * Module constants
+ */
+const GRIDICONS_WITH_DROP = [
+	'add',
+	'cross-circle',
+	'ellipsis-circle',
+	'help',
+	'info',
+	'notice',
+	'pause',
+	'play',
+	'spam',
+];
+
+/**
+ * Constants
+ */
+enum Status {
+	Error = 'is-error',
+	Info = 'is-info',
+	Success = 'is-success',
+	Warning = 'is-warning',
+	Plain = 'is-plain',
+}
+
+interface Props {
+	className?: string;
+	duration?: number;
+	onDismissClick: () => void;
+	translate: ( x: string ) => string;
+	children: React.ReactNode;
+	status?: Status;
+	icon?: string | null;
+	isCompact?: boolean;
+	isLoading?: boolean;
+	showDismiss?: boolean;
+	text?: string | React.ReactNode;
+}
+
+const getIcon = ( status: Status ): string => {
+	let icon: string;
+
+	switch ( status ) {
+		case Status.Info:
+			icon = 'info';
+			break;
+		case Status.Success:
+			icon = 'checkmark';
+			break;
+		case Status.Error:
+			icon = 'notice';
+			break;
+		case Status.Warning:
+			icon = 'notice';
+			break;
+		default:
+			icon = 'info';
+			break;
+	}
+
+	return icon;
+};
+
+const Notice: FunctionComponent< Props > = ( {
+	className = '',
+	duration = 0,
+	onDismissClick,
+	translate,
+	children,
+	status = null,
+	icon = null,
+	isCompact = false,
+	isLoading = false,
+	showDismiss = ! isCompact,
+	text = null,
+} ) => {
+	useEffect( () => {
+		let dismissTimeout: TimerHandle | undefined = undefined;
```

Comment:
`dismissTimeout` won't be persisted between effect calls. Did you mean to create a ref outside the effect to store this in and change in the effect?

Reply:
Thanks @sarayourfriend for chiming in!

That's a good question. Initially I thought that not persisting it would be enough, I changed my mind though f3518bc